### PR TITLE
Set notifications plugin 3.0.0 baseline JDK version to JDK-21

### PR DESCRIPTION
**Pull request by reta**
### Description
Set notifications plugin 3.0.0 baseline JDK version to JDK-21

### Related Issues
Closes https://github.com/opensearch-project/notifications/issues/924

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/notifications/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/core-spi/build.gradle [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import com.github.jengelman.gradle.plugins.shadow.ShadowBasePlugin
 
 plugins {
-    id 'com.github.johnrengelman.shadow'
+    id 'io.github.goooler.shadow' version "8.1.7"
```
Comment (reta) [inline notifications/core-spi/build.gradle:9]: See please https://github.com/opensearch-project/OpenSearch/issues/10745#issuecomment-2183411895

File: notifications/build.gradle [hunk 2]
```diff
@@ -41,6 +42,16 @@ buildscript {
         classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.23.0"
         classpath "org.jacoco:org.jacoco.agent:0.8.11"
     }
+    
+     configurations {
```
Comment (reta) [inline notifications/build.gradle:46]: See please https://github.com/opensearch-project/OpenSearch/issues/10745#issuecomment-2183411895

File: notifications/build.gradle [hunk 3]
```diff
@@ -21,9 +21,10 @@ buildscript {
         }
         opensearch_no_snapshot = opensearch_version.replace("-SNAPSHOT","")
         common_utils_version = System.getProperty("common_utils.version", opensearch_build)
-        kotlin_version = System.getProperty("kotlin.version", "1.8.21")
+        kotlin_version = System.getProperty("kotlin.version", "1.9.25")
```
Comment (yuye-aws) [inline notifications/build.gradle:24]: Why do we need to upgrade the kotlin version? As far as I know, only common-util repo has 1.9.25 as the default kotlin version.

File: notifications/build.gradle [hunk 4]
```diff
@@ -21,9 +21,10 @@ buildscript {
...
         junit_version = System.getProperty("junit.version", "5.7.2")
         aws_version = System.getProperty("aws.version", "1.12.687")
+        asm_version = "9.7"
```
Comment (yuye-aws) [inline notifications/build.gradle:27]: Use `System.getProperty`?

File: notifications/build.gradle (hunk 3 above)
Comment (reta) [reply notifications/build.gradle:24]: Thanks @yuey-aws, afaict Kotlin 1.8 does not support JDK-21 bytecode

**Comment by cwperks** [comment]
Approved the CI to run. Can maintainers of this repo help move this PR forward?

**Review by Hailong-am — APPROVED** [review]
waiting for CI pass to merge

**Pull request merged**